### PR TITLE
Remove unused dependencies

### PR DESCRIPTION
**Pull request by trevor-scheer**
Cleanup unused dependencies.

`dotenv` is only used in the language server, so not needed in vscode.

Types for `react / react-dom` and the `babel-preset-react` were all needed for the webview in vscode, which no longer exists.

This should c_lose #859 as well.
 
TODO:

- [ ] Update CHANGELOG.md\* with your change (include reference to issue & this PR)
- [ ] Make sure all of the significant new logic is covered by tests
- [x] Rebase your changes on master so that they can be merged easily
- [x] Make sure all tests and linter rules pass

\*Make sure changelog entries note which project(s) has been affected. See older entries for examples on what this looks like.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -2342,15 +2317,15 @@
       "dependencies": {
         "@oclif/config": {
           "version": "1.12.10",
-          "bundled": true,
```
Comment (trevor-scheer) [inline package-lock.json]: @abernix what's the verdict on this `bundled/resolved` flapping? I remember you mentioning it, but I don't know if there was resolution. Something with a new version of node maybe?

File: package.json [hunk 2]
```diff
@@ -56,7 +56,6 @@
     "@oclif/test": "1.2.4",
     "@types/babel-types": "7.0.6",
     "@types/babel__generator": "7.0.2",
-    "@types/chai": "4.1.7",
     "@types/common-tags": "1.8.0",
     "@types/cosmiconfig": "5.0.3",
     "@types/globby": "8.0.0",
```
Comment (JakeDawkins) [inline package.json]: I can't find a reference to globby anywhere either. Do you know what this is?

Comment (trevor-scheer) [reply package.json]: Used once upon a time? I killed it 😄

File: package-lock.json (hunk 1 above)
Comment (abernix) [reply package-lock.json]: Switching from `"bundled": true` to `resolved` is great.  The key is make sure you're not committing anything that adds a new `"bundled": true`.  This particular one was _added_ by Renovate, but due to a bug in `npm`.  Renovate is trying to get it ironed out while the npm bug still exists.  See this comment: https://github.com/renovatebot/renovate/issues/3436#issuecomment-475930530

Anyhow, this LGTM.

Comment (trevor-scheer) [reply package-lock.json]: Glad to hear it, and thanks for the info @abernix!

**Pull request merged**